File: Easy/9.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) return false;

        long long revNum = 0;
        int xCopy = x;
        while (x != 0) {
            int lastDigit = x % 10;
            revNum = revNum * 10 + lastDigit;
            x /= 10;
        }

        return (revNum == xCopy ? true : false);
    }
};

File: Easy/9.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        sign = -1 if x < 0 else 1   
        # ternary equivalent to cpp

        revNum = int( str(abs(x))[::-1] )
        # take abs of x, convert to string to slice and reverse, type back to int
        return True if revNum == x else False

# class Solution:
#    def isPalindrome(self, x: int) -> bool:
#        s = str(x)
#        return s == s[::-1]
# also works